src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp

```diff
@@ -238,6 +261,11 @@ namespace Quantum
             typedef unsigned (*TMeasure)(unsigned, unsigned, unsigned*, unsigned*);
             static TMeasure m         = reinterpret_cast<TMeasure>(this->GetProc("Measure"));
             std::vector<unsigned> ids = GetQubitIds(numTargets, targets);
+            if (ids.size() == 1)
+            {
+                // If measuring exactly one qubit, mark it as measured for tracking.
+                isMeasured[ids[0]] = true;
+            }
             return reinterpret_cast<Result>(
                 m(this->simulatorId, (unsigned)numBases, reinterpret_cast<unsigned*>(bases), ids.data()));
         }

```

Comment:
I have an impression that only one qubit is marked as measured regardless of the number of qubits being measured.

Reply:
One classical bit is returned, but that could come from a joint measurement on an arbitrary number of qubits. E.g.: `Measure([PauliZ, PauliZ], [q0, q1])` collapses the state of `[q0, q1]` onto either the +1 or −1 eigenspace of 𝑍 ⊗ 𝑍, spanned by {|00⟩, |11⟩} and {|01⟩, |10⟩}, respectively.

Reply:
If multiple qubits are measured it doesn't guarantee that the individual qubit state is known. Only when the length of `ids` is 1 (ie: when only a single qubit is being measured and not a joint measurement) do we want to mark that qubit as measured and safe to release.

Reply:
Thanks.